translate/storage/lisa.py

```diff
@@ -329,20 +334,45 @@ def serialize_hook(treestring):
     def serialize(self, out=None):
         """Converts to a string containing the file's XML"""
         root = self.document.getroot()
+        xml_quote_format = "'"
+        xml_encoding = self.encoding.lower()
+
         if self.XMLdoublequotes:
-            if self.XMLuppercaseEncoding:
-                out.write(b'<?xml version="1.0" encoding="UTF-8"?>\n')
-            else:
-                out.write(b'<?xml version="1.0" encoding="utf-8"?>\n')
+            xml_quote_format = '"'
+
+        if self.XMLuppercaseEncoding:
+            xml_encoding = self.encoding.upper()
+
+        xml_declaration = f"<?xml version={xml_quote_format}1.0{xml_quote_format} encoding={xml_quote_format}{xml_encoding}{xml_quote_format}?>\n"
+
+        out.write(xml_declaration.encode(self.encoding))
+
         if self.XMLindent:
             reindent(root, **self.XMLindent)
-        treestring = etree.tostring(
-            self.document,
-            pretty_print=not self.XMLindent,
-            xml_declaration=not self.XMLdoublequotes,
-            encoding=self.encoding,
-            doctype=self.XMLdoctype,
-        )
+        # When XMLSelfClosingTags flag is active, relay
+        # against LXML library - otherwise we're using
+        # standard xml library.
+        if self.XMLSelfClosingTags:
+            treestring = etree.tostring(
+                self.document,
+                pretty_print=not self.XMLindent,
+                xml_declaration=False,
+                encoding=self.encoding,
+                doctype=self.XMLdoctype,
+            )
+        else:
+            if self.XMLdoctype:
+                out.write(f"{self.XMLdoctype}\n".encode(self.encoding))
+            stream = io.BytesIO()
+
+            ElementTree.register_namespace("", self.namespace)
+            ElementTree.ElementTree(root).write(
+                stream,
+                self.encoding,
+                short_empty_elements=False,
```

Comment:
This didn't work?

```suggestion
            ElementTree.ElementTree(root).write(
                stream,
                self.encoding,
                short_empty_elements=False,
                default_namespace=self.namespace,
```

Reply:
negative, it seems it fall in a kind of caveat

```
test_xliff.py:671: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
base.py:734: in __bytes__
    self.serialize(out)
xliff.py:870: in serialize
    super().serialize(out)
lisa.py:369: in serialize
    ElementTree.ElementTree(root).write(
C:\Python39\lib\xml\etree\ElementTree.py:746: in write
    qnames, namespaces = _namespaces(self._root, default_namespace)
C:\Python39\lib\xml\etree\ElementTree.py:863: in _namespaces
    add_qname(key)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

qname = 'version'

    def add_qname(qname):
        # calculate serialized qname representation
        try:
            if qname[:1] == "{":
                uri, tag = qname[1:].rsplit("}", 1)
                prefix = namespaces.get(uri)
                if prefix is None:
                    prefix = _namespace_map.get(uri)
                    if prefix is None:
                        prefix = "ns%d" % len(namespaces)
                    if prefix != "xml":
                        namespaces[uri] = prefix
                if prefix:
                    qnames[qname] = "%s:%s" % (prefix, tag)
                else:
                    qnames[qname] = tag # default element
            else:
                if default_namespace:
                    # FIXME: can this be handled in XML 1.0?
>                   raise ValueError(
                        "cannot use non-qualified names with "
                        "default_namespace option"
                        )
E                       ValueError: cannot use non-qualified names with default_namespace option

C:\Python39\lib\xml\etree\ElementTree.py:840: ValueError
```